pkg/utils/helmrepo.go

```diff
@@ -540,3 +556,9 @@ func DeleteHelmReleaseCRD(runtimeClient client.Client, crdx *clientsetx.Clientse
 		}
 	}
 }
+
+//IsURL return true if string is a valid URL
+func IsURL(str string) bool {
+	u, err := url.Parse(str)
```

Comment:
According to this - https://stackoverflow.com/questions/31480710/validate-url-with-standard-package-in-go
it looks url.ParseRequestURI is betten than url.Parse for validaing URL string.
 ```url.ParseRequestURI returns an error if the URL is not valid, not an absolute url, etc etc. url.Parse returns valid on almost anything```

The rest of the fix LGTM